ambry-api/src/main/java/com.github.ambry/network/NetworkSend.java

```diff
@@ -59,7 +51,7 @@ public Send getPayload() {
 
   public void onSendComplete() {
     if (metrics != null) {
-      metrics.updateResponseSendTime(SystemTime.getInstance().milliseconds() - sendStartTimeInMs);
+      metrics.updateSendTime(time.milliseconds() - sendStartTimeInMs);
```

Comment:
Let me try to explain my take here. NetworkSendMetrics has two apis, updateQueueTime(long) and updateSendTime(long). Even I was thinking that we should initialize the starting time when a new NetworkSendMetrics is constructed and when onDequeu(), we should subtract the time and update the queue time. similarly onSendComplete(), should subtract from dequeue time and update the send time. But looks like queueing could have been done way before NetworkSendMetrics are constructed and hence updateQueueTime() takes in a value to be updated. So, I thought I would keep the updateSendTime() to be same lines w/o changing anything.